### PR TITLE
Describe intended audience for this specification

### LINKED ISSUE #32: Indicate intended audience for this document
The changes to ICE candidate gathering are targeted to web browsers; the changes to ICE candidate processing affect all WebRTC endpoints.

### PR DESCRIPTION
**Pull request by juberti**
Also wrap long lines for #processing section.
Fixes #32

### CODE REVIEW

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 1]
```diff
@@ -86,22 +86,32 @@ with dynamically generated Multicast DNS (mDNS) names.
...
+mechanism for WebRTC implementations to register
+ephemeral mDNS {{RFC6762}} names for local private IP addresses, and then
+provide those names, rather than the IP addresses, in the ICE candidates
+they surface to web applications.
+
+Upon receiving ICE candidates with mDNS names, remote clients
+will resolve these names to IP addresses and
+perform ICE processing as usual. In the case where the remote client is a web
+application, the WebRTC implementation will manage this resolution internally
+and will not disclose the actual IP addresses to the application, as above.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: 'as above' is not very clear. Probably refers to mode 2, maybe we can do
s/, as above//

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 2]
```diff
@@ -86,22 +86,32 @@ with dynamically generated Multicast DNS (mDNS) names.
...
+
+While this technique is intended to benefit WebRTC implementations in web
+browsers, by preventing collection of private IP addresses by arbitrary web
+pages, it can also be used by any endpoint that wants to avoid disclosing
+information about its local network to remote peers on other networks.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: Maybe we can split this sentence in two, something like:
"In the case where the remote client is a web
 application, the WebRTC implementation will manage this resolution internally
 and will not disclose the actual IP addresses to the application. This prevents collecting private IP addresses by arbitrary web pages.

This technique can also be used by any endpoint that wants to avoid disclosing information about its local network to remote peers on other networks.
"
As stated in the above comment, I would move the second sentence up.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 3]
```diff
@@ -111,14 +121,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
...
 in Section 7.1 of {{HTMLSpec}}) has its own ICE agent.
 
 ICE Candidate Gathering {#gathering}
-----------------------------
+------------------------------------
+
+This section outlines how mDNS should be used by ICE agents to conceal local
+IP addresses. Naturally, if the ICE agent does not want to conceal its IPs,
+e.g., if it has a priori knowledge that its addresses are in fact public,
+this processing is unnecessary.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: This statement makes sense, it seems related to some other issue though (
https://github.com/youennf/mdns-ice-candidates/issues/31 and PR https://github.com/youennf/mdns-ice-candidates/pull/62).
Maybe we should remove this change there.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 4]
```diff
@@ -155,24 +170,34 @@ identical to those provided during candidate gathering (i.e., MUST NOT
 contain private host IP addresses).
 
 ICE Candidate Processing {#processing}
-----------------------------
+--------------------------------------
+
+This section outlines how received ICE candidates with mDNS names are
+processed by ICE agents, and is relevant to all scenarios.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: s/all scenarios/all WebRTC implementations/?

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 5]
```diff
@@ -86,22 +86,32 @@ with dynamically generated Multicast DNS (mDNS) names.
...
-are not exposed to the web application.
+As detailed in {{IPHandling}}, exposing client private IP addresses by
+default to web applications maximizes the probability of successfully creating
+direct peer-to-peer connections between clients, but creates a significant
+surface for user fingerprinting. {{IPHandling}} recognizes this issue, but also
+admits that there is no current solution to this problem; implementations that
+choose to use Mode 3 to address the privacy concerns often suffer from failing
+or suboptimal connections in WebRTC applications. This is particularly an issue
+on unmanaged networks, typically homes or small offices, where NAT loopback may
+not be supported.
```
[the remark this answers is not in the document]
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: yeah, adding "web applications" caused all the lines below to wrap. Agree that separating this out is probably a good idea, I was thinking of doing a single wrap PR that fixes wrap throughout the doc.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 6]
```diff
@@ -86,22 +86,32 @@ with dynamically generated Multicast DNS (mDNS) names.
...
+
+This document proposes an overall solution to this problem by providing a
+mechanism for WebRTC implementations to register
+ephemeral mDNS {{RFC6762}} names for local private IP addresses, and then
+provide those names, rather than the IP addresses, in the ICE candidates
+they surface to web applications.
```
[the remark this answers is not in the document]
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: The problem is that we can only keep private IPs from web applications. In integrated solutions, the app/platform barrier doesn't exist in the same way, which makes the intent unclear.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 2 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: OK. Where do you think the second sentence should go? I didn't quite follow what you intended.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 3 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: The goal here was to indicate who needs to follow which sections. ICE agents that aren't worried about concealing IPs don't need to follow this section.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 4 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: This is broader than just WebRTC implementations, basically anything that interoperates with WebRTC will probably want to support this, so this could either be all WebRTC-compatible endpoints, or simply 'all implementations'.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 2 above)
Comment (youennf) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: I would reword the two paragraphs as follows:
This document proposes an overall solution to this problem by providing a
mechanism for WebRTC implementations to register ephemeral mDNS {{RFC6762}}
names for local private IP addresses, and then provide those names, rather than
the IP addresses, in the ICE candidates. This technique is intended to benefit
WebRTC implementations in web browsers, by preventing collection of private IP
addresses by arbitrary web pages. It can also be used by any endpoint that wants
to avoid disclosing information about its local network to remote peers on other
networks.

A client that receives an ICE candidate with a mDNS hostname will resolve it
to start establishing a connection. In the case where the client is a web
application, the WebRTC implementation will manage this resolution internally
and will not disclose the actual IP addresses to the application.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 4 above)
Comment (youennf) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Understood.
I like the idea to point out that some implementations might only want to be able to process such ICE candidates while other implementations might also want to generate these candidates.
This is ok to spread this information in the two sections.
Maybe it would read more easily in a single place like the intro.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Fixed

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 2 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Took a shot at this.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 4 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Added some text to the intro; reiterated it briefly here.

**Comment by juberti** [comment]
Comments addressed, PTAL.

**Comment by juberti** [comment]
Will wait for @youennf approval before merging.
